lib/clients/authenticated.js

```diff
@@ -1,332 +1,240 @@
+  }
 
-  prototype.getAccountHistory = function(accountID, args, callback) {
-    var self = this;
-
-    args = args || {}
-    if (!callback && (typeof args === 'function')) {
+  getAccountHistory(accountID, args = {}, callback) {
+    if (!callback && typeof args == 'function') {
       callback = args;
       args = {};
     }
 
-    var opts = { 'qs': args };
-    return prototype.get.call(self, ['accounts', accountID, 'ledger'], opts, callback);
-  };
-
-  prototype.getAccountHolds = function(accountID, args, callback) {
-    var self = this;
+    return this.get(['accounts', accountID, 'ledger'], { qs: args }, callback);
+  }
 
-    args = args || {}
-    if (!callback && (typeof args === 'function')) {
+  getAccountHolds(accountID, args = {}, callback) {
+    if (!callback && typeof args == 'function') {
       callback = args;
       args = {};
     }
 
-    var opts = { 'qs': args };
-    return prototype.get.call(self, ['accounts', accountID, 'holds'], opts, callback);
-  };
+    return this.get(['accounts', accountID, 'holds'], { qs: args }, callback);
+  }
 
-  prototype._placeOrder = function(params, callback) {
-    var self = this;
+  _placeOrder(params, callback) {
+    let requiredParams = ['size', 'side', 'product_id'];
 
-    var requiredParams = ['size', 'side', 'product_id'];
+    if (params.type !== 'market') requiredParams.push('price');
 
-    if (params.type !== 'market') {
-      requiredParams.push('price');
-    }
+    this._requireParams(params, requiredParams);
 
-    _.forEach(requiredParams, function(param) {
-      if (params[param] === undefined) {
-        throw "`opts` must include param `" + param + "`";
-      }
-    });
-    var opts = { 'body': params };
-    return prototype.post.call(self, ['orders'], opts, callback);
-  };
+    return this.post(['orders'], { body: params }, callback);
+  }
 
-  prototype.buy = function(params, callback) {
-    var self = this;
+  buy(params, callback) {
     params.side = 'buy';
-    return self._placeOrder(params, callback);
-  };
+    return this._placeOrder(params, callback);
+  }
 
-  prototype.sell = function(params, callback) {
-    var self = this;
+  sell(params, callback) {
     params.side = 'sell';
-    return self._placeOrder(params, callback);
-  };
+    return this._placeOrder(params, callback);
+  }
 
-  prototype.getTrailingVolume = function(callback) {
-    var self = this;
-    return prototype.get.call(self, ['users', 'self', 'trailing-volume'], {}, callback);
-  };
+  getTrailingVolume(callback) {
+    return this.get(['users', 'self', 'trailing-volume'], {}, callback);
+  }
 
-  prototype.cancelOrder = function(orderID, callback) {
-    var self = this;
+  cancelOrder(orderID, callback) {
 
-    if (!callback && (typeof orderID === 'function')) {
-      callback = orderID;
-      callback(new Error('must provide an orderID or consider cancelOrders'));
-      return;
+    if(!orderID || typeof orderID == 'function') {
```

Comment:
I think their original intent was to ensure that the first argument was an orderID instead of a callback. So if `orderID` were a function, then `!orderID === true`, so you have to also check to make sure it's not a function.